dali/core/endian_util_test.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <dali/core/endian_util.h>
+#include <dali/core/int_literals.h>
+
+namespace dali {
+
+TEST(EndianUtilTest, Fundamental) {
+  int32_t a = 0x12345678;
+  swap_endian(a);
+  EXPECT_EQ(a, 0x78563412);
+  uint64_t b = 0x0123456789abcdef_u64;
+  swap_endian(b);
+  EXPECT_EQ(b, 0xefcdab8967452301_u64);
+  char c = 'x';
+  EXPECT_EQ(c, 'x');
+  uint32_t f_ieee_swapped = 0x0000803f;
+  float f = *(const float *)&f_ieee_swapped;
+  swap_endian(f);
+  EXPECT_EQ(f, 1);
+}
+
+namespace {
+
+struct Inner {
+  int16_t x, y;
+};
+
+struct Outer {
+  Inner inner;
+  int a[2];
+  std::array<int, 2> b;
+};
+
+}  // namespace
+
+SWAP_ENDIAN_FIELDS(Inner, x, y);
```

Comment:
I understand that preprocessor magic has its own rules, but it was not obvious to me what this does. The name suggests that it will "swap endian fields", but it *creates a function which can be used to* swap endian fields. Maybe some short doxygen comment would help?

Reply:
I've added documentation.